assets/src/edit-story/components/form/numeric.js

```diff
@@ -97,48 +111,124 @@ function Numeric({
   canBeEmpty,
   ...rest
 }) {
-  const isMultiple = value === MULTIPLE_VALUE;
-  const placeholder = isMultiple ? __('multiple', 'web-stories') : '';
-  const [dot, setDot] = useState(false);
   const [inputValue, setInputValue] = useState(value);
-  const ref = useRef();
 
-  useEffect(() => setInputValue(value), [value]);
+  const inputRef = useRef();
+  const targetFormRef = useRef(null);
+  const skipValidationRef = useRef(false);
+
+  const { focused, handleFocus, handleBlur } = useFocusAndSelect(inputRef);
+
+  const isMultiple = inputValue === MULTIPLE_VALUE;
+  const placeholder = isMultiple ? __('multiple', 'web-stories') : '';
+
+  const handleChange = useCallback((event) => {
+    setInputValue(event.target.value);
+  }, []);
+
+  const validateAndSubmitInput = useCallback(() => {
+    targetFormRef.current = inputRef.current.form;
+
+    const validValue = validateInput(inputValue, {
+      canBeNegative,
+      canBeEmpty,
+      float,
+    });
+
+    setInputValue(validValue ?? value);
+    onChange(validValue ?? value);
+  }, [inputValue, canBeNegative, canBeEmpty, float, value, onChange]);
 
   const handleUpDown = useCallback(
     ({ key, altKey }) => {
       if (isMultiple) {
         return;
       }
 
-      let newValue;
+      const validValue = validateInput(inputValue, {
+        canBeNegative,
+        canBeEmpty,
+        float,
+      });
+
+      // The `|| 0` is to prevent newValue from being set to
+      // empty string by either validValue or value
+      let newValue = (validValue ?? value) || 0;
       const diff = Big(float && altKey ? 0.1 : 1);
       if (key === 'ArrowUp') {
         // Increment value
         newValue =
           typeof max !== 'undefined'
-            ? Math.min(max, Big(value).plus(diff))
-            : Big(value).plus(diff);
+            ? Math.min(max, Big(newValue).plus(diff))
+            : Big(newValue).plus(diff);
       } else if (key === 'ArrowDown') {
         // Decrement value
         newValue =
           typeof min !== 'undefined'
-            ? Math.max(min, Big(value).minus(diff))
-            : Big(value).minus(diff);
+            ? Math.max(min, Big(newValue).minus(diff))
+            : Big(newValue).minus(diff);
+      }
+
+      setInputValue(newValue.toString());
```

Comment:
I think we should submit the value here as well and select the entire input field contents. It does pollute the history quite a bit with a bunch of entries, but that also happens when you select an element and move it with arrow keys, so that's on par.

At least until we implement some history compacting local previously postponed in #67.